test/unit/model/consent.test.ts

```diff
@@ -172,7 +193,71 @@ describe('src/model/consent', (): void => {
 
       expect(consents[0].id).toEqual(partialConsent.id)
       expect(consents[0].createdAt).toEqual(expect.any(String))
-      expect(consents[0]).toEqual(expect.objectContaining(completeConsent))
+      expect(consents[0]).toEqual(expect.objectContaining(consentWithOnlyUpdateFields))
+    })
+
+    it('updates existing consent with only non-conflicting fields from a consent', async (): Promise<void> => {
+      // Inserting record to update
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+
+      // Action
+      const updateCount: number = await consentDB.update(conflictingConsent)
+
+      expect(updateCount).toEqual(1)
+
+      // Assertion
+      const consents: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: completeConsent.id
+        })
+
+      expect(consents[0].createdAt).toEqual(expect.any(String))
+
+      // Conflicting fields (initiatorId and participantId) are still the same
+      expect(consents[0]).toEqual(expect.objectContaining(partialConsent))
```

Comment:
Take a look at `https://jestjs.io/docs/en/expect#expectobjectcontainingobject` to give you more flexibility in checking your test assertions.